### DISCUSSION TITLE
form-floating accessibility?

### DISCUSSION THREAD

**Discussion by wingradirect** [twbs/bootstrap#39343] [Q&A] [answered]
Are there any accessibility concerns when the label for an input comes after the input, like all of the examples for the form-floating feature? I really like how the fields are presented, but can't afford to sacrifice accessibility and put ourselves at risk of lawsuits.

**Comment by julien-deramond** [comment]
@patrickhlauke could confirm but I don't think there are any accessibility concerns with the form floating feature when the label for input comes after the input since they are "linked" with the `id`/`for` attributes.

**Comment by patrickhlauke** [comment, marked as answer]
The one quirk with these is that if you're just using virtual cursor to read though the page, you will get a slightly odd order (matching the source order), but yes when navigating in forms mode, the form controls will be announced correctly based on their associated labels.

Randomly, was thinking that for the next iteration of Bootstrap, we could experiment with using `:has()` and the parent container to work around the need for the unusual sibling selector that requires this particular back-to-front ordering...

**Reply by patrickhlauke** [reply]
for fun, I had a little play with how the `:has()` approach could look like https://codepen.io/patrickhlauke/pen/ExrNLzN

**Reply by wingradirect** [reply]
Thank you for confirming "when navigating in forms mode, the form controls will be announced correctly"
